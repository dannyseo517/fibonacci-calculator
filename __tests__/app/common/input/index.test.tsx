import React from 'react';
import { shallow } from 'enzyme';
import Input from 'Common/input';

describe('Common input', () => {
    it('should render initial layout', () => {
        const callback = (): any => {
            return null;
        };
        const component = shallow(<Input placeholder='placeholder test' callback={callback} autofocus={true} />);
        expect(component.getElements()).toMatchSnapshot();
    });

    it('should respond to input change and update state', () => {
        const callback = jest.fn();
        const component = shallow(<Input placeholder='placeholder test' callback={callback} autofocus={true} />);
        component.find('input').simulate('change', { target: { value: '10' } });
        expect(component.state('value')).toEqual('10');
    });
});

