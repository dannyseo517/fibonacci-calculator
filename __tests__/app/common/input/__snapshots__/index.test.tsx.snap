// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common input should render initial layout 1`] = `
Array [
  <input
    autoFocus={true}
    onChange={[Function]}
    placeholder="placeholder test"
    value=""
  />,
]
`;
